torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -48,12 +49,18 @@ void removeTupleNodes(Node* n, bool must_remove_tuples) {
       }
       return;
     }
-    n->output()->replaceAllUsesWith(construct->inputs().at(*maybe_int));
+    auto tuple = n->inputs().at(0);
+    const size_t tuple_len = tuple->type()->containedTypes().size();
+    const size_t normalized_idx = normalizeIndex(*maybe_int, tuple_len);
+    n->output()->replaceAllUsesWith(construct->inputs().at(normalized_idx));
```

Comment:
I think this is the only change needed, but it's not safe to use the normalized_idx since it maybe out of bounds. I don't think tupleslice has a problem.